driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java

```diff
@@ -286,6 +289,8 @@ private BsonDocument getCommandToEncode() {
                 addReadConcernDocument(extraElements, sessionContext);
             }
             extraElements.add(new BsonElement("autocommit", BsonBoolean.FALSE));
+        } else if (sessionContext.isSnapshot()) {
+            addReadConcernDocument(extraElements, sessionContext);
```

Comment:
Same pattern we use for transactions (just above).  Higher level code defers adding read concern for snapshot reads, allowing it to be appended in a central location (here).